odl/solvers/functional/default_functionals.py

```diff
@@ -2470,6 +2470,149 @@ def __repr__(self):
+    Wikipedia article: https://en.wikipedia.org/wiki/Bregman_divergence
+
+    [Bur2016] Burger, M. *Bregman Distances in Inverse Problems and Partial
+    Differential Equation*. In: Advances in Mathematical Modeling, Optimization
+    and Optimal Control, 2016. p. 3-33. Arxiv version online at
+    https://arxiv.org/abs/1505.05191
+    """
+
+    def __init__(self, functional, point, subgradient_op=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        functional : `Functional`
+            The functional on which to base the Bregman distance. If the
+            optional argument `subgradient_op` is not given, the functional
+            needs to implement `functional.gradient`.
+        point : element of ``functional.domain``
+            The point from which to define the Bregman distance
+        subgradient_op : `Operator`, optional
+            The operator that takes an element in `functional.domain` and
+            returns a subgradient of the functional in that point.
+            Default: ``functional.gradient``.
+
+        Examples
+        --------
+        Example of initializing the Bregman distance functional:
+
+        >>> space = odl.uniform_discr(0, 2, 14)
+        >>> l2_squared = odl.solvers.L2NormSquared(space)
+        >>> point = space.one()
+        >>> Bregman_dist = odl.solvers.BregmanDistance(l2_squared, point)
+
+        This is gives the shifted L2 norm squared ||x - 1||:
+
+        >>> Bregman_dist(space.zero())
+        2.0
+        """
+        if not isinstance(functional, Functional):
+            raise TypeError('`functional` {} not an instance of ``Functional``'
+                            ''.format(functional))
+        self.__functional = functional
+
+        if point not in functional.domain:
+            raise ValueError('`point` {} is not in `functional.domain` {}'
+                             ''.format(point, functional.domain))
+        self.__point = point
+
+        if subgradient_op is None:
+            # If the subgradient is not given, assign the gradient of the
+            # functional
+            try:
+                self.__subgradient_op = self.__functional.gradient
+            except NotImplementedError:
+                raise NotImplementedError('`subgradient` not given, and the '
+                                          '`functional` {} does not implement '
+                                          '`functional.gradient`'
+                                          ''.format(functional))
+        else:
+            # Check that given subgradient is an operator that maps from the
+            # domain of the functional to the domain of the functional
+            if not isinstance(subgradient_op, Operator):
+                raise TypeError('`subgradient_op` {} is not an instance of '
+                                '``Operator``'.format(subgradient_op))
+            if not self.__functional.domain == subgradient_op.domain:
```

Comment:
prefer

```
if (thing to be tested) == (expected value)
```

it more closely follows human language so it's easier to read

Reply:
And generally, it is preferable to write `(something) != (something else)` instead of `not (something) == (something else)`. For example pylint will complain about this.